src/syscheckd/create_db.c

```diff
@@ -1574,6 +1591,83 @@ void fim_diff_folder_size(){
     os_free(diff_local);
 }
 
+void update_wildcards_config(OSList *directories,
+                             OSList *wildcards){
+    OSListNode *node_it;
+    OSListNode *aux_it;
+    directory_t *dir_it;
+    directory_t *new_entry;
+    char **paths;
+
+    if (wildcards == NULL || directories == NULL) {
+        return;
+    }
+
+    mdebug2(FIM_WILDCARDS_UPDATE_START);
+    w_rwlock_wrlock(&syscheck.directories_lock);
+
+    OSList_foreach(node_it, directories) {
+        dir_it = node_it->data;
+        dir_it->is_expanded = 0;
+    }
+
+    OSList_foreach(node_it, wildcards) {
+        dir_it = node_it->data;
+        paths = expand_wildcards(dir_it->path);
+
+        if (paths == NULL) {
+            continue;
+        }
+
+        for (int i = 0; paths[i]; i++) {
+            new_entry = fim_copy_directory(dir_it);
+            os_free(new_entry->path);
+            new_entry->path = paths[i];
+            new_entry->is_expanded = 1;
+#ifndef WIN32
+            if (CHECK_FOLLOW & new_entry->options) {
+                new_entry->symbolic_links = realpath(new_entry->path, NULL);
+            }
+#endif
+            if (new_entry->diff_size_limit == -1) {
+                new_entry->diff_size_limit = syscheck.file_size_limit;
+            }
+
+            if (FIM_MODE(new_entry->options) == FIM_WHODATA) {
+#ifdef WIN32
+                realtime_adddir(new_entry->path, new_entry, 0);
+#else
+                add_whodata_directory(new_entry->path);
+#endif
+            }
```

Comment:
I'm not 100% sure this calls should be made here, the directory should be scanned prior to it being placed in whodata mode.